packages/core/bootstrap/src/lib/ws/epics.ts

```diff
@@ -124,21 +131,22 @@ export const connectEpic: Epic<AnyAction, AnyAction, any, any> = (action$, state
             )
             .pipe(
               withLatestFrom(state$),
-              map(([message, state]) => {
-                // TODO: Use merge map to not discard the first message
+              mergeMap(([message, state]) => {
                 if (!state.ws.subscriptions[subscriptionKey]?.active) {
-                  return subscribed(payload)
+                  return of(subscribed(payload), messageReceived({ message, subscriptionKey })).pipe(
+                    log(`WS: New subscription ${JSON.stringify(payload.subscriptionMsg)}`),
+                    take(1),
```

Comment:
Why `take(1)` here? This will only let the first `subscribed` through, and discard the next `messageReceived`.

I suggest removing this line.

Reply:
Yep right, I wanted the log to be printed only for the first event

Reply:
Yeah, this will not do it!

You would need to `subscribed(payload).pipe(log....)`, but mixing core logic with logs becomes overwhelming at one point. I suggest you move logs to a separate pipeline altogether, as I did with metrics. Metrics are handled in a completely separate pipeline.

Reply:
Opening an issue on ws logging